torch/_subclasses/fake_tensor.py

```diff
@@ -1002,6 +1008,9 @@ def _prep_args_for_hash(self, args: Any) -> Any:
                     raise _BypassDispatchCache("constant attribute")
                 if arg.is_sparse:
                     raise _BypassDispatchCache("sparse tensor")
+                # sparse tensors don't have storage, so check is after
+                if isinstance(arg.untyped_storage().nbytes(), torch.SymInt):
```

Comment:
Last change - FakeTensor caching doesnt support dynamic shapes, and it looks like you can have a tensor with static sizes/strides/offset, but with a dynamic storage nbytes.